downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/ProtegeExporter.java

```diff
@@ -0,0 +1,404 @@
+							{
+								LocalDateTime exportStart = LocalDateTime.now();
+								process = processBuilder.start();
+								// Now, wait for protege export to complete.
+								int exitCode = process.waitFor();
+								LocalDateTime exportEnd = LocalDateTime.now();
+								logger.info("Finished generating protege files for {}; Elapsed time: {}; Exit code is {}", pathway.toString(), Duration.between(exportStart, exportEnd), exitCode);
+								if (exitCode == 0)
+								{
+									Files.createDirectories(Paths.get("/tmp/protege_files"));
+									// Move the file to the protege_files, overwrite existing.
+									Files.move(Paths.get("/tmp/"+fileName+".tar.gz"), Paths.get(PROTEGE_FILES_DIR+fileName+".tar.gz"), StandardCopyOption.REPLACE_EXISTING);
+								}
+								else
+								{
+									// Should a non-zero exit throw an exception and interrupt the whole process? I am leaning to "no" - if a *single* pathway fails to export,
+									// I think I'd want the rest of them to continue, and then we can re-run for just the failed pathway.
+									logger.error("Non-zero exit code from protegeexporter script: {}. Check logs for more details.", exitCode);
+								}
+							}
+							catch (IOException e)
+							{
+								e.printStackTrace();
+							}
+							catch (InterruptedException e)
+							{
+								e.printStackTrace();
+								// If something interrupts, force the process to terminate.
+								process.destroyForcibly();
+							}
+						}
+					});
+				}).get();
+				LocalDateTime overallEnd = LocalDateTime.now();
+				pool.shutdown();
+				logger.info("Overall elapsed time: {}", Duration.between(overallStart, overallEnd));
+			}
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Determine if a pathway should be processed. A pathway should be processed IF: <ul>
+	 * <li>An ID filter has been specified and the pathway's ID is in that set.</li>
+	 * <li>A species filter has been specified and the pathway's species is in that set.</li></ul>
+	 * If an ID filter has NOT been specified, then no filtering by ID will occurr. If a species filter has NOT be specified, then no filtering by species will occur.
+	 * If NO filters are specified, then this function will return true for any pathway. 
+	 * @param idFilterInEffect
+	 * @param speciesFilterInEffect
+	 * @param pathway
+	 * @return
+	 */
+	private boolean shouldPathwayBeProcessed(final boolean idFilterInEffect, final boolean speciesFilterInEffect, GKInstance pathway)
+	{
+		boolean processPathway = true;
+		if (idFilterInEffect)
+		{
+			if (!this.pathwayIdsToProcess.contains(pathway.getDBID()))
+			{
+				processPathway = false;
+			}
+		}
+		if (speciesFilterInEffect)
+		{
+			try
+			{
+				String pathwaySpecies = ((GKInstance) pathway.getAttributeValue("species")).getDisplayName();
+				if (!this.speciesToProcess.contains(pathwaySpecies))
+				{
+					processPathway = false;
+				}
+			}
+			catch (Exception e)
+			{
+				e.printStackTrace();
```

Comment:
If trying to obtain the pathwaySpecies throws an exception, should `processPathway` be set to false in the catch block?  I would guess that if the pathwaySpecies couldn't be obtained, the pathway shouldn't be processed with the `speciesFilterInEffect` being true?

Reply:
Could the exception condition be logged, also?